subxt/src/events/events_type.rs

```diff
@@ -759,7 +771,7 @@ mod tests {
         }
 
         // Create fake metadata that knows about our single event, above:
-        let metadata = metadata::<Event>();
+        let metadata = Arc::new(RwLock::new(metadata::<Event>()));
```

Comment:
This pattern of creating metadata in an Arc + RwLock is copied a lot; perhaps a helper function (`metadata()`?) should handle the arc/rwlocking too?

Reply:
Have changed it! Thanks!